### ISSUE TITLE
.in filter with empty set creates invalid SQL

### ISSUE THREAD

**Issue by bygri** [vapor/fluent#158]
When Fluent creates a scope/set filter (Filter.swift, line 103) with an empty set:

```swift
let query = try DummyModel.query().filter("id", .in, [])
```

it returns the following SQL:

```sql
SELECT `dummy_models`.* FROM `dummy_models` WHERE `dummy_models`.`id` IN ()
```

This is not valid, in MySQL at least and causes a syntax error. Ideally, calling this function with an `.in` scope and an empty set should short-circuit and result in an empty query that, when executed, doesn't return anything. Calling with a `.notIn` scope and an empty set should act as if no filter had been called, and not add to the `WHERE` clause.

**Comment by BrettRToomey** [comment]
@tanner0101 @LoganWright Has this been fixed?

**Comment by mcdappdev** [comment]
I'm still seeing this issue in Vapor 2.9

**Comment by fwgreen** [comment]
If the 2.0 API documentation is correct, then `subset` might not be fully implemented:
```Swift
try query.filter("id", in: [1, 2])
```
```
error: argument labels '(_:, in:)' do not match any available overloads
```

**Comment by vzsg** [comment]
For the record, it's not correct: vapor/documentation#174

**Comment by mcdappdev** [comment]
I can get it to work if I append a -1 to the array since that'll never actually be in an array of ids so that there's always something to look at

**Comment by tanner0101** [comment]
@vzsg I made issue https://github.com/vapor/fluent/issues/269 for that. Those overloads should exist.

**Comment by vzsg** [comment]
You have my sword.

**Comment by fwgreen** [comment]
@tanner0101 I'm still experiencing this issue after rebuilding with 2.0.3
```
[PostgreSQL Error: ERROR:  syntax error at or near ")"
LINE 1: SELECT "persons".* FROM "persons" WHERE "persons"."id" IN ()
                                                                   ^
] [Identifier: PostgreSQL.PostgreSQLError.42601 (syntaxError)] [Suggested Fixes: Fix the invalid syntax in your query, If an ORM has generated this error, report the issue to its GitHub page]
```

**Comment by tanner0101** [comment]
@fwgreen it was fixed in 2.0.4
